op-node/rollup/driver/sync_deriver.go

```diff
@@ -104,24 +104,38 @@ func (s *SyncDeriver) OnEvent(ctx context.Context, ev event.Event) bool {
 }
 
 func (s *SyncDeriver) OnUnsafeL2Payload(ctx context.Context, envelope *eth.ExecutionPayloadEnvelope) {
-	// If we are doing CL sync or done with engine syncing, fallback to the unsafe payload queue & CL P2P sync.
-	if s.SyncCfg.SyncMode == sync.CLSync || !s.Engine.IsEngineSyncing() {
-		s.Log.Info("Optimistically queueing unsafe L2 execution payload", "id", envelope.ExecutionPayload.ID())
+	s.Log.Debug("OnUnsafeL2Payload called", "id", envelope.ExecutionPayload.ID(), "syncMode", s.SyncCfg.SyncMode, "followUnsafeWithRRSync", s.SyncCfg.SyncModeReqResp)
+
+	// Scenario: syncMode.ELSync with s.SyncCfg.SyncModeReqResp=false.
+	//
+	// 1. For the conditional: !s.Engine.IsEngineInitialELSyncing() && s.SyncCfg.SyncModeReqResp ->
+	//    When initial EL Sync is done, conditional will not be satisfied. The unsafe payload will be directly pushed to EL using InsertUnsafePayload.
+	//
+	// 2. For the conditional: !s.Engine.IsEngineInitialELSyncing() ->
+	//    When initial EL Sync is done, conditional will be satisfied, pushing the unsafe payload to the queue. At checkForGapInUnsafeQueue, it will pop the unsafe payload and apply InsertUnsafePayload because s.SyncCfg.SyncModeReqResp=false.
+	//
+	// Both conditionals will lead to calling InsertUnsafePayload, but if we don't include the `&& s.SyncCfg.SyncModeReqResp`, there may be a delay before the payload is applied, at most the interval of `altSyncTicker(syncCheckInterval := time.Duration(s.SyncDeriver.Config.BlockTime) * time.Second * 2)`.
+	if s.SyncCfg.SyncMode == sync.CLSync || (!s.Engine.IsEngineInitialELSyncing() && s.SyncCfg.SyncModeReqResp) {
 		s.Engine.AddUnsafePayload(ctx, envelope)
-	} else if s.SyncCfg.SyncMode == sync.ELSync {
+		return
+	}
+
+	if s.SyncCfg.SyncMode == sync.ELSync {
 		ref, err := derive.PayloadToBlockRef(s.Config, envelope.ExecutionPayload)
 		if err != nil {
-			s.Log.Info("Failed to turn execution payload into a block ref", "id", envelope.ExecutionPayload.ID(), "err", err)
+			s.Log.Warn("Failed to turn execution payload into a block ref", "id", envelope.ExecutionPayload.ID(), "err", err)
```

Comment:
This should arguably even be an error log

```suggestion
			s.Log.Error("Failed to turn execution payload into a block ref", "id", envelope.ExecutionPayload.ID(), "err", err)
```